src/Sarif/HashUtilities.cs

```diff
@@ -7,11 +7,35 @@
 
 namespace Microsoft.CodeAnalysis.Sarif
 {
+    public class HashData
+    {
+        string md5;
+        string sha1;
+        string sha256;
+
+        public HashData()
+        {
+            this.md5 = String.Empty;
+            this.sha1 = String.Empty;
+            this.sha256 = String.Empty;
+        }
+
+        public HashData(string md5, string sha1, string sha256)
+        {
+            this.md5 = md5;
+            this.sha1 = sha1;
+            this.sha256 = sha256;
+        }
+
+        public string GetMD5 { get { return md5; } }
+        public string GetSha1 { get { return sha1; } }
+        public string GetSha256 { get { return sha256; } }
+
+    }
```

Comment:
The only place this class is constructed is in `FileData` (the other file, where you changed the arguments to the method call). There is no place that default-constructs a `HashData`.

Also notice that the properties of `HashData` are read-only ("get"), so if you default-constructed one, its properties would always be empty. What would such an object represent? If you don't have hash data available, you can just return a null instead of a `HashData`, and that's actually what `FileData` does.

This is a specific example of a general principle, called "YAGNI", which means "You Ain't Gonna Need It". We try not to write a single line of code until it's needed. Today, I can't see a need for the default constructor, so we don't write one.